Python/specialize.c

```diff
@@ -2413,18 +2414,19 @@ _Py_Specialize_CompareOp(_PyStackRef lhs_st, _PyStackRef rhs_st, _Py_CODEUNIT *i
             goto failure;
         }
         else {
-            instr->op.code = COMPARE_OP_STR;
+            specialized_op = COMPARE_OP_STR;
             goto success;
         }
     }
     SPECIALIZATION_FAIL(COMPARE_OP, compare_op_fail_kind(lhs, rhs));
 failure:
     STAT_INC(COMPARE_OP, failure);
-    instr->op.code = COMPARE_OP;
+    SET_OPCODE_OR_RETURN(instr, COMPARE_OP);
```

Comment:
The same applies to that piece of code, you can remove both of failure and success blocks, now you just need to do `unspecialize(instr, compare_op_fail_kind(lhs, rhs))`. And in fact there's no more a `SET_OPCODE_OR_RETURN` macro :)

Also don't forget to remove the  `#ifdef Py_STATS` directive near the `compare_op_fail_kind`.